dbms/src/Interpreters/Join.cpp

```diff
@@ -424,15 +425,14 @@ void Join::insertFromBlock(const Block & block, size_t stream_index)
         bool force_spill_partition_blocks = false;
         {
             std::unique_lock lk(build_probe_mutex);
-            if (max_bytes_before_external_join && bytes_to_be_added + getTotalByteCount() >= max_bytes_before_external_join)
+            if (hash_join_spill_context->getOperatorSpillThreshold() > 0 && bytes_to_be_added + getTotalByteCount() >= hash_join_spill_context->getOperatorSpillThreshold())
```

Comment:
Maybe this code should move to `HashJoinSpillContext::updatePartitionRevocableMemory`

Reply:
But it needs to estimate the extra bytes brought by this block(before split), not very easy to move to `HashJoinSpillContext::updatePartitionRevocableMemory`

Reply:
Why do we need to estimate memory used before spill?
Can totoal_memory_used be maintained in SpillContext to determine whether to spill?

Reply:
I've update the code to remove `force_spill_partition_blocks`